rules/DowngradePhp70/Rector/FunctionLike/DowngradeThrowableTypeDeclarationRector.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DowngradePhp70\Rector\FunctionLike;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\ArrowFunction;
+use PhpParser\Node\Expr\Closure;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Function_;
+use PHPStan\Type\ObjectType;
+use Rector\BetterPhpDocParser\PhpDocParser\PhpDocFromTypeDeclarationDecorator;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\DowngradePhp70\Rector\FunctionLike\DowngradeThrowableTypeDeclarationRector\DowngradeThrowableTypeDeclarationRectorTest
+ */
+final class DowngradeThrowableTypeDeclarationRector extends AbstractRector
+{
+    public function __construct(
+        private readonly PhpDocFromTypeDeclarationDecorator $phpDocFromTypeDeclarationDecorator,
+    ) {
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Function_::class, ClassMethod::class, Closure::class, ArrowFunction::class];
```

Comment:
could you remove `ArrowFunction::class` and its fixtures? actually, `ArrowFunction` need to be converted to `Closure` first via:

https://github.com/rectorphp/rector-src/blob/fee8a47d1a627b01bfba2d85293c84692715506a/rules/DowngradePhp74/Rector/ArrowFunction/ArrowFunctionToAnonymousFunctionRector.php#L21

to work on php <=7.3, that's to avoid double call on target closure node.

Reply:
Sure. How does it actually work when multiple rectors match a single node. Does it run a random one repeatedly until convergence?

Reply:
that's loop until file not has changed:

https://github.com/rectorphp/rector-src/blob/fee8a47d1a627b01bfba2d85293c84692715506a/src/Application/FileProcessor/PhpFileProcessor.php#L63-L85

with stop on 9 counter to avoid infinite loop, you can add 1 more new test class+config+fixture with double register service to test if it is working:

```php
<?php

declare(strict_types=1);

use Rector\DowngradePhp70\Rector\FunctionLike\DowngradeThrowableTypeDeclarationRector;
use Rector\DowngradePhp74\Rector\ArrowFunction\ArrowFunctionToAnonymousFunctionRector;
use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;

return static function (ContainerConfigurator $containerConfigurator): void {
    $services = $containerConfigurator->services();
    $services->set(ArrowFunctionToAnonymousFunctionRector::class);
    $services->set(DowngradeThrowableTypeDeclarationRector::class);    
};
```

For multiple rules applied check, you can add under `tests/Issues` https://github.com/rectorphp/rector-src/tree/main/tests/Issues